src/providers/spatialite/qgsspatialitedataitemguiprovider.cpp

```diff
@@ -134,7 +136,9 @@ void QgsSpatiaLiteDataItemGuiProvider::deleteConnection( QgsDataItem *item )
                               QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) != QMessageBox::Yes )
     return;
 
-  QgsSpatiaLiteConnection::deleteConnection( item->name() );
+  QgsProviderMetadata *providerMetadata = QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "spatialite" ) );
+  providerMetadata->deleteConnection( item->name() );
+
```

Comment:
QgsSpatiaLiteDataItemGuiProvider::createDatabase should be updated to use providerMetadata->saveConnection instead too